torch/_subclasses/fake_tensor.py

```diff
@@ -333,6 +335,18 @@ def from_real_tensor(
         constraint_dims: "Optional[DimList[DimConstraint]]" = None,
         memoized_only=False,
     ):
+        if self.parent is not None:
+            t = self.parent.from_real_tensor(
+                fake_mode.parent,
+                t,
+                make_constant=make_constant,
+                shape_env=shape_env,
+                ignore_subclass=ignore_subclass,
+                source=source,
+                dynamic_dims=dynamic_dims,
+                constraint_dims=constraint_dims,
+                memoized_only=memoized_only,
+            )
```

Comment:
I guess one (non-obvious) invariant now is that if you create a `FakeTensorMode` with a parent, then its `tensor_memo` cache will always be key'd on `FakeTensor`s instead of real tensors. I guess there's nothing wrong with that? (mostly just calling it out)

Reply:
Yep! In fact, we actually do this already, when you run _dynamo.export with inputs that are already fake tensors (e.g., because you used fake tensors to avoid loading the model for tracing)